### DISCUSSION TITLE
Windows 8 to 8.1 upgrade

### DISCUSSION THREAD

**Discussion by Plasti-pews** [LegacyUpdate/LegacyUpdate#254] [Questions & Help]
When installing Legacy Update on Windows 8, it offers to upgrade to Windows 8.1, but when selecting this option, it just redirects to Microsoft's page where the upgrade is no longer available. Is there any way to actually upgrade from 8 to 8.1 anymore? You used to have to do it through the Store, but you can't do it that way anymore.

**Comment by kirb** [comment]
This is definitely an [annoying problem](https://hachyderm.io/@kirb/112744803224373969), I’ll need to write up a page on the Legacy Update site to replace the Microsoft download page. There’s the possibility that the Windows Update-based upgrade still exists (this was released much later, after Windows 10, likely based on the Get Windows 10 app), but I don’t quite understand what conditions are required to invoke it. If that’s not viable I may just have to suggest you find a copy yourself (I’d rather not directly link to a rehost).

**Reply by Plasti-pews** [reply]
The worst part is that I've seen claims that Microsoft updated their ISOs to allow using Windows 8 keys on Windows 8.1 installation media, but I have NEVER managed to get that to work. I have two Windows 8 keys from when I had a Technet subscription that I could easily use, and they work fine for installing 8, but won't work for any 8.1 installations. I've tried finding the update file itself, but Microsoft made that VERY difficult. Their search always wants to return one of the later updates for 8.1, not the upgrade from 8 to 8.1. I thought I had that file saved somewhere (and probably do along with some other KB files), but I can't even find a listing for the KB number for it. Do you happen to know that info? It would certainly help with my search for this elusive software. Maybe even putting the KB up on Archive.org would be helpful for people who are struggling to find it.

**Reply by kirb** [reply]
Unfortunately not sure about that because I only own a Windows 8.1 key, not 8, but I feel like a long time ago I did try using the key to install 8.0, and it did let me use it (I didn’t let it activate though, didn’t want to use up a new hardware config on that key). Was pretty sure at least retail keys are interchangeable, similar to how Windows 7/8/8.1 keys would pass during Windows 10 setup, and activate in the full OS. But I also seem to recall the Store upgrade method was required (you couldn’t just download the matching 8.1 iso from the download page) because it did something to make your key *eligible* for activating 8.1.

It’s possible the way licensing works with the Windows Update method, *if* it’s based on Get Windows 10, leaves activation to occur after the upgrade to 8.1. Unfortunately, it’s also possible the “downlevel” activation path was removed at the same time it was removed for Windows 10/11 [last year](https://www.windowscentral.com/software-apps/windows-11/microsoft-kills-loophole-that-let-windows-7-keys-activate-windows-11-and-10). It will be a little difficult to test that this path still works, because it requires a Windows 8 key that has never been used to upgrade to Windows 8.1.

The update itself appears to be [KB3008273](https://support.microsoft.com/en-us/topic/an-update-to-enable-an-automatic-update-from-windows-8-to-windows-8-1-8358d5ae-a774-ab7b-3171-17d3c07dd462). I’ll look further into it soon.

**Reply by sal-ammoniac** [reply]
> This is definitely an [annoying problem](https://hachyderm.io/@kirb/112744803224373969), I’ll need to write up a page on the Legacy Update site to replace the Microsoft download page. There’s the possibility that the Windows Update-based upgrade still exists (this was released much later, after Windows 10, likely based on the Get Windows 10 app), but I don’t quite understand what conditions are required to invoke it. If that’s not viable I may just have to suggest you find a copy yourself (I’d rather not directly link to a rehost).

The upgrade cannot be done now. Microsoft removed most Windows 8.1 ESDs from WU server. So user must use iso now.

**Reply by kirb** [reply]
I wrote a longer message here that I then lost... the short version is that I looked through all the domains 8.1 ESDs were hosted from (variations of `dl.ws.microsoft.com`), and all are completely gone now. IPs/CNAMEs from historic DNS records don't manage to return anything either. It's definitely some left over Windows 8 Store infrastructure that's now fully gone away.

It seems there is still a trick to get the iframe from the old 8.1 download page to serve you signed links for 8.1 Update 3. A piracy site I won't mention/link to here uses this trick to let you grab the legit original isos. Looking at it, the way it works entirely seems like a bug. I don't think I want to rely on that.

Unfortunately I think there is just no option for us to provide either a feature to automatically upgrade 8 to 8.1, or instructions for you to do so manually.

Also to confirm from my prior message, 8 and 8.1 share product keys/activation, so at least we don't need to worry about losing Windows Store servers that provide the upgrade license.
